verification/cbmc/proofs/aws_cryptosdk_priv_try_gen_key/aws_cryptosdk_priv_try_gen_key_harness.c

```diff
@@ -0,0 +1,246 @@
+                                                 .decrypt_materials = nondet_voidp() };
+    __CPROVER_assume(aws_cryptosdk_cmm_vtable_is_valid(&vtable));
+
+    struct aws_cryptosdk_cmm *cmm = malloc(sizeof(*cmm));
+    __CPROVER_assume(cmm);
+    cmm->vtable = &vtable;
+    __CPROVER_assume(aws_cryptosdk_cmm_base_is_valid(cmm));
+    session->cmm = cmm;
+
+    enum aws_cryptosdk_alg_id alg_id;
+    struct aws_cryptosdk_alg_properties *props = aws_cryptosdk_alg_props(alg_id);
+    __CPROVER_assume(aws_cryptosdk_alg_properties_is_valid(props));
+    session->alg_props = props;
+
+    struct aws_cryptosdk_hdr *hdr = ensure_nondet_hdr_has_allocated_members(MAX_TABLE_SIZE);
+
+    __CPROVER_assume(aws_cryptosdk_hdr_members_are_bounded(hdr, MAX_EDK_LIST_ITEMS, MAX_BUFFER_SIZE));
+
+    __CPROVER_assume(IMPLIES(hdr != NULL, aws_byte_buf_is_bounded(&hdr->iv, session->alg_props->iv_len)));
+    __CPROVER_assume(IMPLIES(hdr != NULL, aws_byte_buf_is_bounded(&hdr->auth_tag, session->alg_props->tag_len)));
+
+    ensure_cryptosdk_edk_list_has_allocated_list_elements(&hdr->edk_list);
+
+    __CPROVER_assume(aws_cryptosdk_hdr_is_valid(hdr));
+
+    // The header should have been cleared earlier
+    __CPROVER_assume(aws_array_list_length(&hdr->edk_list) == 0);
+
+    __CPROVER_assume(hdr->enc_ctx.p_impl != NULL);
+    ensure_hash_table_has_valid_destroy_functions(&hdr->enc_ctx);
+    size_t empty_slot_idx;
+    __CPROVER_assume(aws_hash_table_has_an_empty_slot(&hdr->enc_ctx, &empty_slot_idx));
+    session->header = *hdr;
+
+    struct aws_array_list *keyring_trace = malloc(sizeof(*keyring_trace));
+    __CPROVER_assume(keyring_trace != NULL);
+    __CPROVER_assume(aws_array_list_is_bounded(
+        keyring_trace, MAX_TRACE_NUM_ITEMS, sizeof(struct aws_cryptosdk_keyring_trace_record)));
+    __CPROVER_assume(keyring_trace->item_size == sizeof(struct aws_cryptosdk_keyring_trace_record));
+    ensure_array_list_has_allocated_data_member(keyring_trace);
+    __CPROVER_assume(aws_array_list_is_valid(keyring_trace));
+    ensure_trace_has_allocated_records(keyring_trace, MAX_STRING_LEN);
+    __CPROVER_assume(aws_cryptosdk_keyring_trace_is_valid(keyring_trace));
+    session->keyring_trace = *keyring_trace;
+
+    session->alloc = can_fail_allocator();
+    __CPROVER_assume(aws_allocator_is_valid(session->alloc));
+    __CPROVER_assume(session->alloc != NULL);
+    session->header.edk_list.alloc = session->alloc;  // this assumption is needed for build_header
+    __CPROVER_assume(aws_cryptosdk_commitment_policy_is_valid(session->commitment_policy));
+
+    __CPROVER_assume(session->state == ST_GEN_KEY);
+    __CPROVER_assume(session->mode == AWS_CRYPTOSDK_ENCRYPT);
+
+    struct content_key *content_key = malloc(sizeof(*content_key));
+    __CPROVER_assume(content_key != NULL);
+    session->content_key = *content_key;
+
+    /* Function under verification */
+    aws_cryptosdk_priv_try_gen_key(session);
```

Comment:
This still requires postconditions as you mention in the PR description.

Reply:
Yep, they are added now along with precondition on the source code.